File: python/sensor_utils.py
from electronic_sensor import ElectronicSensor

def calculate_efficiency(reading, max_scale):
    efficiency = (reading / max_scale) * 100.0
    return efficiency


def filter_critical(assets):
    is_critical = lambda a: isinstance(a, ElectronicSensor) and a.current_reading > 75.0
    return list(filter(is_critical, assets))

File: java/SensorUtils.java
import java.util.List;
import java.util.stream.Collectors;

public class SensorUtils {

    public static double calculateEfficiency(double reading, double maxScale) {
        double efficiency = (reading / maxScale) * 100.0; // local variable
        return efficiency;
    }

    public static List<ElectronicSensor> filterCritical(List<IndustrialAsset> assets) {
        return assets.stream()
                .filter(a -> a instanceof ElectronicSensor)
                .map(a -> (ElectronicSensor) a)
                .filter(s -> s.getCurrentReading() > 75.0)
                .collect(Collectors.toList());
    }
}
